src/wallet/wallet.cpp

```diff
@@ -1842,12 +1846,16 @@ int CWallet::ScanForWalletTransactions(CBlockIndex* pindexStart, bool fUpdate)
                     ret++;
             }
 
-            ZCIncrementalMerkleTree tree;
+            ZCIncrementalMerkleTree sproutTree;
+            ZCSaplingIncrementalMerkleTree saplingTree;
             // This should never fail: we should always be able to get the tree
             // state on the path to the tip of our chain
-            assert(pcoinsTip->GetSproutAnchorAt(pindex->hashSproutAnchor, tree));
+            assert(pcoinsTip->GetSproutAnchorAt(pindex->hashSproutAnchor, sproutTree));
+            if (pindex->pprev) {
```

Comment:
@Eirik0 Double-checking that we need this conditional `if (pindex->pprev)` ?

Reply:
This was added based on a comment from @str4d:
https://github.com/zcash/zcash/pull/3353#discussion_r201297006
I would like to defer to him for the answer to that question.

Reply:
@str4d ^ ... also, could you re-review with a focus on changes to state on disk?  I think it's fine but would benefit from another look.